File: examples/scala/micro/UserService.scala
package micro

case class UserInfo(id: Option[String], token: String)

class UserService {
  private val authService = new AuthService()

  def getUser(token: String): UserInfo = {
    val userId = authService.verifyToken(token)
    UserInfo(id = userId, token = token)
  }

  def updateUser(token: String, action: String): Option[String] = {
    val userId = authService.verifyToken(token)
    userId.foreach { uid =>
      authService.runAdminCommand(uid, action)
    }
    userId
  }
}

File: examples/swift/micro/UserService.swift
import Foundation

struct UserInfo {
    let id: String?
    let token: String
}

class UserService {
    let authService = AuthService()

    func getUser(token: String) -> UserInfo {
        let userId = authService.verifyToken(token)
        return UserInfo(id: userId, token: token)
    }

    func updateUser(token: String, action: String) -> String? {
        let userId = authService.verifyToken(token)
        if let uid = userId {
            authService.runAdminCommand(uid, action)
        }
        return userId
    }
}
